webapp/src/pages/EditAvatarPage/EditAvatarPage.tsx

```diff
@@ -11,15 +12,72 @@ import {
 } from '../../shared/components';
 import { USERS_EP_URL, WILDCARD_AVATAR_URL } from '../../shared/urls';
 import { goBack } from '../../shared/callbacks';
-import { useCallback } from 'react';
+import React, { useCallback, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
-import { EditableAvatar } from '../../shared/components/Avatar/EditableAvatar';
+import { usersApi } from '../../shared/services/ApiService';
+import { useDrag } from '../../shared/hooks/UseDrag';
 import { useAuth } from '../../shared/hooks/UseAuth';
 
+export const EDITABLE_AVATAR_SCALE = 1.29;
+export const EDITABLE_AVATAR_SCALE_REVERSE = 1 / EDITABLE_AVATAR_SCALE;
+
+type ImgData = {
+  imgHash: number;
+  imgName?: string;
+  imgFile: File | null;
+};
+
 export default function EditAvatarPage() {
-  const submitChanges = useCallback(() => true, []);
   const { user } = useAuth();
   const navigate = useNavigate();
+  const { picturePosition, handleMouseDown, handleMouseMove, handleMouseUp } =
+    useDrag(user ? { x: user.avatarX, y: user.avatarY } : null);
+  const [imgData, setImgData] = useState<ImgData>({
+    imgHash: Date.now(),
+    imgFile: null,
+  });
+
+  const { imgName, imgHash, imgFile } = imgData;
+  const submitPlacement = useCallback(async () => {
+    usersApi
+      .userControllerUpdateCurrentUserRaw({
+        updateUserDto: {
+          avatarX: Math.round(
+            picturePosition.x * EDITABLE_AVATAR_SCALE_REVERSE,
+          ),
+          avatarY: Math.round(
+            picturePosition.y * EDITABLE_AVATAR_SCALE_REVERSE,
+          ),
+        },
+      })
+      .catch((e) => console.error(e));
+  }, [picturePosition]);
+
+  const uploadAvatar = useCallback(async () => {
+    if (imgFile !== null) {
+      usersApi
+        .userControllerUploadAvatar({ file: imgFile })
+        .catch((e) => console.error(e))
+        .finally(() => {
+          setImgData({
+            imgName: undefined,
+            imgHash: Date.now(),
+            imgFile: null,
+          });
+        });
+    }
+  }, [imgFile]);
+
+  const changeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event &&
+      event?.target &&
+      event?.target?.files?.[0] &&
+      setImgData({
+        imgFile: event?.target?.files[0],
+        imgName: event?.target?.files[0]?.name ?? null,
```

Comment:
I have to get the falsy types for imgName coherent. I have a small mess now.

Reply:
suggestion: We could do something like this:

```tsx
    if (event.target.files === null) {
      return;
    }

    setImgData({
      imgFile: event.target.files[0],
      imgName: event.target.files[0].name,
      imgHash: Date.now().toString(),
    });
```

Reply:
done